02-express-tutorial/app.js

```diff
@@ -1,53 +1,20 @@
 const express = require('express');
 const app = express();
-const port = 3000;
-const { products } = require('./data');
+require('dotenv').config();
+const port = process.env.PORT || 3000;
 
-app.get('/', (req, res) => {
-  res.send('<h1>Home Page</h1><a href="/api/products">products</a>');
-});
-app.get('/api/products', (req, res) => {
-  const newProducts = products.map((product) => {
-    const { id, name, image } = product;
-    return { id, name, image };
-  });
-  res.json(newProducts);
-});
-
-app.get('/api/products/:productID', (req, res) => {
-  const { productID } = req.params;
-  const singleProduct = products.find(
-    (product) => product.id === Number(productID),
-  );
-  if (!singleProduct) {
-    return res.status(404).send('Product Does Not Exist');
-  }
-  return res.json(singleProduct);
-});
+const people = require('./routes/people');
+const auth = require('./routes/auth');
+//static assets
+app.use(express.static('./methods-public'));
+// parse form data;
+app.use(express.urlencoded({ extended: false }));
 
-app.get('/api/products/:productID/reviews/:reviewID', (req, res) => {
-  console.log(req.params);
-  res.send('Hello World');
-});
+// parse.json
+app.use(express.json());
```

Comment:
Why do we need this? What does it do?

Reply:
app.use() - sets up middleware that will be executed every time a request is made to the server. This middleware is responsible for parsing incoming JSON payloads, converting them into JavaScript objects, and making them available in the req.body property, which can be accessed and used in the route handler. Using this middleware is very important because without it, the server wouldn't be able to parse JSON data, and the application wouldn't be able to access the data sent by the client.